pmm-app/tests/qan_test.js

```diff
@@ -148,15 +150,131 @@ xScenario('Open the QAN Dashboard and check Details -> Tables', async (I, adminP
+  const filterCheckboxSelector = '#query-analytics-filters input[type="checkbox"]';
+  I.amOnPage(qanPage.url);
+  I.waitForElement(filterCheckboxSelector, 30);
+  I.checkOption(filterCheckboxSelector);
+  I.waitForResponse('http://localhost/v0/qan/Filters/Get', 10);
+  I.click(qanPage.elements.resetAllButton);
+  I.dontSeeCheckboxIsChecked(filterCheckboxSelector);
+});
+
+Scenario('Open the QAN Dashboard, add a column and make sure that this option not available anymore @new-qan', async (I, adminPage, qanPage) => {
+  const addColumnDropdownFirstElement = 'ul.ant-select-dropdown-menu li:first-child';
+  I.amOnPage(qanPage.url);
+  I.waitForElement(qanPage.fields.addColumnsSelector, 30);
+  I.click(qanPage.fields.addColumnsSelector);
+  let labelName = await I.grabAttributeFrom(addColumnDropdownFirstElement, 'label');
+  let selectedLabel = `ul.ant-select-dropdown-menu li[label='${labelName}']`;
+  I.seeElement(selectedLabel);
+  I.click(addColumnDropdownFirstElement);
+  I.click(qanPage.fields.addColumnsSelector);
+  I.dontSeeElement(selectedLabel);
+});
+
+Scenario('Open the QAN Dashboard and verify that hovering over a time metric displays a tooltip with a graph @new-qan', async (I, adminPage, qanPage) => {
+  const queryTimeColumn = '.ant-table-tbody tr:first-child td:last-child';
+  I.amOnPage(qanPage.url);
+  I.waitForElement(`${queryTimeColumn} .summarize`, 30);
+  I.scrollTo(queryTimeColumn);
+  I.moveCursorTo(`${queryTimeColumn} .summarize`);
+  I.seeElement(qanPage.elements.metricTooltip);
+  I.seeElement(qanPage.elements.latencyChart);
+});
+
+Scenario('Open the QAN Dashboard and verify that hovering over a non-time metric displays a tooltip without a graph @new-qan', async (I, adminPage, qanPage) => {
+  const queryCountColumn = '.ant-table-tbody tr:first-child td:nth-child(4)';
+  I.amOnPage(qanPage.url);
+  I.waitForElement(`${queryCountColumn} .summarize`, 30);
+  I.scrollTo(queryCountColumn);
+  I.moveCursorTo(`${queryCountColumn} .summarize`);
+  I.seeElement(qanPage.elements.metricTooltip);
+  I.dontSeeElement(qanPage.elements.latencyChart);
+});
+
+Scenario('Open the QAN Dashboard and verify that the metric value matches the "Per sec" value in the tooltip. @new-qan', async (I, adminPage, qanPage) => {
+  const queryCountColumn = '.ant-table-tbody tr:first-child td:nth-child(4)';
+  I.amOnPage(qanPage.url);
+  I.waitForElement(`${queryCountColumn} .summarize`, 30);
+  I.scrollTo(queryCountColumn);
+  I.moveCursorTo(`${queryCountColumn} .summarize`);
+  I.waitForElement(qanPage.elements.metricTooltip, 5);
+  I.seeElement(qanPage.elements.metricTooltip);
+  let qpsMetricValue = await I.grabTextFrom(`${queryCountColumn} .summarize`);
+  let qpsTooltipValue = await I.grabTextFrom('[data-qa="metrics-list"] [data-qa="qps"] span');
+  assert.equal(qpsMetricValue.replace(/[^0-9.]/g,""), qpsTooltipValue.replace(/[^0-9.]/g,""));
+});
```

Comment:
I.scrollTo is required, because otherwise column on the right side of table is not visible on 13'' screen.
I presume, there is no much sense to move assertion to qanPage.js, because it won't  work unless we move other test code either. But if we move everything it won't increase readability.

Reply:
keeping hardcoded locators in test is a bad practice - qpsTooltipValue locator should be in qanPage.js
  actions like string grabbing and manipulation should be in pageObj
  actually this assertion could be reused for different tests
  so please move 203-205 lines to separate method on qanPage.js called verifyMetricsMatch(firstMetricLocator, secondMetricLocator)